packages/datetime/src/dateInput.tsx

```diff
@@ -198,15 +200,29 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
         };
     }
 
+    public componentWillUnmount() {
+        this.unregisterPopoverBlurHandler();
+        super.componentWillUnmount();
+    }
+
     public render() {
         const { value, valueString } = this.state;
         const dateString = this.state.isInputFocused ? valueString : this.getDateString(value);
         const date = this.state.isInputFocused ? this.createMoment(valueString) : value;
         const dateValue = this.isMomentValidAndInRange(value) ? fromMomentToDate(value) : null;
+        const dayPickerProps = {
+            ...this.props.dayPickerProps,
+            onMonthChange: () => setTimeout(this.registerPopoverBlurHandler, 0),
```

Comment:
why the timeout here? leave a code comment explaining why it's necessary

Reply:
also, should `safeInvoke(this.props.dayPickerProps.onMonthChange)` in here, or suddenly that one prop is not supported.

Reply:
- [ ] add why timeout is required as comment
- [ ] `safeInvoke(this.props.dayPickerProps.onMonthChange)`